pkg/server/server.go

```diff
@@ -269,6 +279,48 @@ func WithMaxAuthorizationModelSizeInBytes(size int) OpenFGAServiceV1Option {
 	}
 }
 
+// WithRateLimitedCheckResolverEnabled sets whether dispatch throttling is enabled
+func WithRateLimitedCheckResolverEnabled(enabled bool) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.rateLimitedCheckResolverEnabled = enabled
+	}
+}
+
+// WithRateLimitedCheckResolverTimerTickerFrequency sets how frequent dispatch throttling is enabled
+func WithRateLimitedCheckResolverTimerTickerFrequency(frequency time.Duration) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.rateLimitedCheckResolverTimerTickerFrequency = frequency
+	}
+}
+
+// WithRateLimitedCheckResolverLowPriorityLevel sets the number of dispatches a request incur before being placed on low priority shaper
+func WithRateLimitedCheckResolverLowPriorityLevel(level uint32) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.rateLimitedCheckResolverLowPriorityLevel = level
+	}
+}
+
+// WithRateLimitedCheckResolverLowPriorityShaper sets the number of tickers required to dispatch a low priority work
+func WithRateLimitedCheckResolverLowPriorityShaper(shaper uint32) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.rateLimitedCheckResolverLowPriorityShaper = shaper
+	}
+}
+
+// WithRateLimitedCheckResolverMediumPriorityLevel sets the number of dispatches a request incur before being placed on medium priority shaper
```

Comment:
Can you explain here what is the difference between the low and the medium priority shaper?

And why is there no "high" priority shaper?

Reply:
High priority means no shaping.  Perhaps we relabel medium priority to high priority?